101-asp-net-app-service/azuredeploy.json

```diff
@@ -0,0 +1,134 @@
+{
+    "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "name": {
+            "type": "string",
+            "metadata":{
+                "description": "That name is the name of our application. It has to be unique.Type a name followed by your resource group name. (<name>-<resourceGroupName>)"
+            }
+        },
+        "location": {
+            "type": "string",
+            "allowedValues":["Central US", "North Europe", "West Europe", "Southeast Asia", "East Asia", "West US", "East US", "Japan West", "Japan East", "East US 2", 
+            "North Central US", "South Central US", "Brazil South", "Australia East", "Australia Southeast", "Central India", "West India", "South India", "Canada Central",
+            "Canada East", "West Central US", "West US 2", "UK West", "UK South", "Korea South", "Korea Central", "France South", "France Central", "Australia Central 2", 
+            "Australia Central", "South Africa North", "South Africa West", "Switzerland North", "Germany West Central"], 
+            "metadata":{
+                "description": "Location where the App Service Plan will be deployed."
+            }
+        }
+    },
+    "variables": {
+        "subscriptionId": "[subscription().subscriptionId]",
+        "hostingEnvironment": "",
+        "serverFarmResourceGroup": "[resourceGroup().name]",
+        "alwaysOn": false,
+        "sku": "Free",
+        "skuCode": "F1",
+        "workerSize": "0",
+        "workerSizeId": 0,
+        "numberOfWorkers": "1",
+        "currentStack": "dotnet",
+        "netFrameworkVersion": "v4.0",
+        "hostingPlanName": "[concat(parameters('name'), '-hpn')]"
+    },
+    "resources": [
+        {
+            "apiVersion": "2018-02-01",
+            "name": "[parameters('name')]",
+            "type": "Microsoft.Web/sites",
+            "location": "[parameters('location')]",
+            "tags": null,
+            "dependsOn": [
+                "[concat('microsoft.insights/components/', parameters('name'), '-insights')]",

```

Comment:
use a variable for the insights/components name since you reuse it multiple times